### PR TITLE
Temporarily fix UDP egress for zoned clusters

### PR DESCRIPTION
**Pull request by tim-ebert**
**What this PR does / why we need it**:
With this PR the azure extensions add a dummy service of type `LoadBalancer` with a UDP port to the Shoot's `kube-system` namespace (zoned Shoots only).
This allows UDP egress from the shoot workers.

This is only a temporary workaround and should be removed once we have a proper implementation for UDP egress via Azure NAT gateways.

**Which issue(s) this PR fixes**:
ref #1 

**Special notes for your reviewer**:
cc @rfranzke @dkistner 

**Release note**:

```improvement operator
UDP egress is now enabled for zoned Azure clusters via a temporary workaround. For this the Azure Provider extension installs an additional Service of Type `LoadBalancer` called `udp-egress-dummy` with a single UDP port into the `kube-system` Namespace of the Shoot.
```

### CODE REVIEW

File: charts/internal/controlplane-shoot/charts/udp-egress-dummy/templates/udp-egress-dummy.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# this dummy service creates a LoadBalancer with a UDP port to enable outgoing UPD traffic for Shoots using
+# Azure Standard LoadBalancers (see https://github.com/gardener/gardener-extension-provider-azure/issues/1)
+# TODO (dkistner): this workaround should be removed once we have a proper implementation for UDP egress via NAT GW
+apiVersion: v1
+kind: Service
+metadata:
+  name: udp-egress-dummy
+  namespace: kube-system
+  labels:
+    app: dummy
```
Comment (rfranzke) [inline charts/internal/controlplane-shoot/charts/udp-egress-dummy/templates/udp-egress-dummy.yaml]: Can you add an annotation and explain why we create this? People might see this service/try to delete it.

File: pkg/controller/controlplane/valuesprovider.go [hunk 2]
```diff
@@ -116,6 +116,17 @@ var ccmShootChart = &chart.Chart{
 	},
 }
 
+var udpEgressDummyChart = &chart.Chart{
+	Name: "udp-egress-dummy",
+	Path: filepath.Join(azure.InternalChartsPath, "udp-egress-dummy"),
```
Comment (rfranzke) [inline pkg/controller/controlplane/valuesprovider.go]: This path looks wrong. Why have we moved the things into a `controlplane-shoot` subfolder in the first place?

File: charts/internal/controlplane-shoot/charts/udp-egress-dummy/templates/udp-egress-dummy.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{{ if .Values.enabled }}
+# this dummy service creates a LoadBalancer with a UDP port to enable outgoing UPD traffic for Shoots using
+# Azure Standard LoadBalancers (see https://github.com/gardener/gardener-extension-provider-azure/issues/1)
+# TODO (dkistner): this workaround should be removed once we have a proper implementation for UDP egress via NAT GW
```
[the remark this answers is not in the document]
Comment (dkistner) [reply charts/internal/controlplane-shoot/charts/udp-egress-dummy/templates/udp-egress-dummy.yaml]: +1

**Comment by tim-ebert** [comment]
Sure, will do.
I also want to test my changes again before we merge this, but I'm stuck with some errors in my local setup...

**Comment by tim-ebert** [comment]
I have now tested the changes in my local setup.
Creation and cleanup of the LoadBalancer Service and Azure LoadBalancer rules for UDP work fine and is only done for zoned clusters.
I also verified that NTP timesync is working properly with the fix in place.

So this PR should be fine now, PTAL :)

**Pull request merged**